packages/jest-matchers/src/__tests__/__snapshots__/matchers-test.js.snap

```diff
@@ -400,6 +400,20 @@ Expected value to be undefined, instead received
   [31m{}[39m"
 `;
 
+exports[`.toBeDefined(), .toBeUndefined() '{}' is defined 3`] = `
```

Comment:
the reason why this happened is because snapshot didn't work well with duplicated titles.

this is the test
<img width="553" alt="screen shot 2016-09-16 at 12 42 14 pm" src="https://cloud.githubusercontent.com/assets/940133/18599082/5268c4d4-7c0b-11e6-8f0e-6a1cc5a3c036.png">

and these are the titles:
<img width="285" alt="screen shot 2016-09-16 at 12 42 08 pm" src="https://cloud.githubusercontent.com/assets/940133/18599103/6df409fc-7c0b-11e6-8b59-7a54b7bfc537.png">

as you can see `{} is defined` is duplicated (new Map() is serialized to {}, we need to fix that btw)

the previous implementation of snapshots would overwrite the existing ones, and the implementation in this PR will keep incrementing the counters